### ISSUE TITLE
Initializing Kitchen::RakeTasks causes error if Vagrant not installed

### ISSUE THREAD

**Issue by mfischer-zd** [test-kitchen/test-kitchen#645]
If vagrant is not installed or not available in $PATH, initializing Kitchen::RakeTasks will cause an uncaught exception to be thrown, even if the merged configuration (after KITCHEN_LOCAL_YAML is set) doesn't use a vagrant driver.

```
Kitchen::UserError: Vagrant 1.1.0 or higher is not installed. Please download a package from http://www.vagrantup.com/downloads.html.
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:325:in `rescue in vagrant_version'
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:323:in `vagrant_version'
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:148:in `verify_dependencies'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/driver.rb:45:in `for_plugin'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:226:in `new_driver'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:240:in `new_instance'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:129:in `block in build_instances'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:128:in `map'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:128:in `with_index'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:128:in `build_instances'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:104:in `instances'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/rake_tasks.rb:51:in `block in define'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/rake_tasks.rb:50:in `define'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/rake_tasks.rb:37:in `initialize'
vendor/bundle/ruby/2.1.0/bundler/gems/zendesk_chef_support-5c39fced5af7/lib/zendesk_chef_support/tasks/kitchen.rb:13:in `new'
vendor/bundle/ruby/2.1.0/bundler/gems/zendesk_chef_support-5c39fced5af7/lib/zendesk_chef_support/tasks/kitchen.rb:13:in `<top (required)>'
vendor/bundle/ruby/2.1.0/bundler/gems/zendesk_chef_support-5c39fced5af7/lib/zendesk_chef_support/tasks.rb:8:in `require_relative'
vendor/bundle/ruby/2.1.0/bundler/gems/zendesk_chef_support-5c39fced5af7/lib/zendesk_chef_support/tasks.rb:8:in `<top (required)>'
Rakefile:2:in `require'
Rakefile:2:in `<top (required)>'
Errno::ENOENT: No such file or directory - vagrant
vendor/bundle/ruby/2.1.0/gems/mixlib-shellout-1.6.1/lib/mixlib/shellout/unix.rb:294:in `exec'
vendor/bundle/ruby/2.1.0/gems/mixlib-shellout-1.6.1/lib/mixlib/shellout/unix.rb:294:in `block in fork_subprocess'
vendor/bundle/ruby/2.1.0/gems/mixlib-shellout-1.6.1/lib/mixlib/shellout/unix.rb:276:in `fork'
vendor/bundle/ruby/2.1.0/gems/mixlib-shellout-1.6.1/lib/mixlib/shellout/unix.rb:276:in `fork_subprocess'
vendor/bundle/ruby/2.1.0/gems/mixlib-shellout-1.6.1/lib/mixlib/shellout/unix.rb:52:in `run_command'
vendor/bundle/ruby/2.1.0/gems/mixlib-shellout-1.6.1/lib/mixlib/shellout.rb:242:in `run_command'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/shell_out.rb:64:in `run_command'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/driver/base.rb:182:in `run_command'
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:250:in `run'
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:270:in `run_silently'
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:322:in `vagrant_version'
vendor/bundle/ruby/2.1.0/gems/kitchen-vagrant-0.16.0/lib/kitchen/driver/vagrant.rb:148:in `verify_dependencies'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/driver.rb:45:in `for_plugin'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:226:in `new_driver'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:240:in `new_instance'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:129:in `block in build_instances'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:128:in `map'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:128:in `with_index'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:128:in `build_instances'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/config.rb:104:in `instances'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/rake_tasks.rb:51:in `block in define'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/rake_tasks.rb:50:in `define'
vendor/bundle/ruby/2.1.0/gems/test-kitchen-1.3.1/lib/kitchen/rake_tasks.rb:37:in `initialize
```

**Comment by fnichol** [comment]
Hey @mfischer-zd, would a better error message in kitchen-vagrant help make this clearer? In effect, kitchen-vagrant wouldn't be able to complete its task as it runs the Vagrant binary without a fully qualified path. On the other hand, I may have misunderstood :smile:

**Comment by mfischer-zd** [comment]
I don't understand why an error would need to be thrown in the first place.  After all, test-kitchen isn't using Vagrant when the configuration is resolved per the current environment:

```
  project_config:
    filename: ".kitchen.yml"
    raw_data:
      driver:
        name: vagrant
      provisioner:
        name: chef_zero
        require_chef_omnibus: 11.18.0
        data_bags_path: test/integration/default/data_bags
        roles_path: test/integration/default/roles
        attributes:
          testing: true
      platforms:
      - name: centos-7.0
      suites:
      - name: default
        run_list:
        - recipe[xxxxxx]
  local_config:
    filename: ".kitchen.cloud.yml"
    raw_data:
      driver:
        name: ec2
      platforms:
      - name: centos-7.0
        driver_config:
          image_id: ami-xxxxxx
          username: centos
          availability_zone: us-east-1a
          aws_access_key_id: key_id
          aws_secret_access_key_id: access_key
          aws_ssh_key_id: ssh_key_id
          flavor_id: m3.large
          security_group_ids:
          - sg-xxxxxxx
  combined_config:
    filename: 
    raw_data:
      driver:
        name: ec2
      provisioner:
        name: chef_zero
        require_chef_omnibus: 11.18.0
        data_bags_path: test/integration/default/data_bags
        roles_path: test/integration/default/roles
        attributes:
          testing: true
      platforms:
      - name: centos-7.0
        driver_config:
          image_id: ami-xxxxxxx
          username: centos
          availability_zone: us-east-1a
          aws_access_key_id: key_id
          aws_secret_access_key_id: access_key
          aws_ssh_key_id: ssh_key_id
          flavor_id: m3.large
          security_group_ids:
          - sg-xxxxxxx
      suites:
      - name: default
        run_list:
        - recipe[xxxxxx]
```

**Comment by tyler-ball** [comment]
I think I understand what you are saying.  The vagrant Rake tasks are supper aggressive with loading dependencies, so it fails to load the vagrant executable even if you eventually end up not needing it.  Am I understanding that correctly?

**Comment by tyler-ball** [comment]
We theorize that [this](https://github.com/test-kitchen/test-kitchen/blob/master/lib/kitchen/rake_tasks.rb#L34) could be lazier (or we need to shortcut around them) because its the Config object that checks for Vagrant.

**Comment by mfischer-zd** [comment]
Any update on this?

**Comment by fsimmend** [comment]
we just faked vagrant installation in bundlers bin dir.

**Comment by martinisoft** [comment]
Hitting this error now trying to use rake tasks with Travis-CI. Despite us overriding the YAML and changing drivers to something not vagrant, it still insists on vagrant being present. Trying to also change KITCHEN_YAML has no effect either so my hunch is that despite having a preferred driver, it continues to load the vagrant one anyway if it happens to be in the project config.

**Comment by martinisoft** [comment]
So a way to re-create this issue is to do the following:

* Have a project yaml as `.kitchen.yml` with a vagrant driver
* Create a `.kitchen.cloud.yml` with the ec2 or digitalocean driver and set the `KITCHEN_LOCAL_YAML=.kitchen.cloud.yml` env var
* Attempt to use a rake task on a system without vagrant installed. Watch it explode.
* Remove the `.kitchen.yml` and rename the `.kitchen.cloud.yml` to `.kitchen.yml` that now contains the non-vagrant driver
* Watch in (possible) amazement as your rake tasks work again

So my hunch based on the above steps is the rake task invocation is ignoring the computed YAML file and just going with the default project one anyway.

**Comment by smurawski** [comment]
Thanks @martinisoft - investigating.

**Comment by martinisoft** [comment]
@smurawski Awesome! As a side note, we invoked the tasks in our Rakefile with:

```ruby
require 'kitchen/rake_tasks'
Kitchen::RakeTasks.new
```

**Comment by smurawski** [comment]
Ah ha.. The environmental variables for directing which YAML doc are only interpreted for the CLI, not the rake task. https://github.com/test-kitchen/test-kitchen/blob/c228aa12ee0447881ff3d226088700ec47e1d877/lib/kitchen/cli.rb#L73

That's a pretty straight forward fix.

**Comment by smurawski** [comment]
Closed by #1046 and shipping in v1.9.1 shortly
